command-line-api-2/Program.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.CommandLine;
+using System.CommandLine.Builder;
+using System.CommandLine.Invocation;
+using System.CommandLine.Parsing;
+using System.Threading.Tasks;
+
+public class Program
+{
+    public string String { get; set; }
+    public bool Bool { get; set; }
+
+    static void Run(bool b, string s)
+    {
+        Console.WriteLine($"Bool option: {b}");
+        Console.WriteLine($"String option: {s}");
+    }
+
+    private static async Task<int> Main(string[] args)
+    {
+        RootCommand command = new RootCommand();
+        command.AddOption(new Option<bool>(new[] { "--bool", "-b" }, "Bool option"));
+        command.AddOption(new Option<string>(new[] { "--string", "-s" }, "String option"));
+
+        command.Handler = CommandHandler.Create<bool, string>(Run);
```

Comment:
This is non-realistic. The typical programs have dozens of options. Passing several dozens of options as generic arguments does not work.

Reply:
We support binding complex classes, so the command option/argument count isn't constrained by the handler method parameter count. It's similar to ASP.NET Core model binding in this regard.

Reply:
I updated the PR. Performance between these two iterations is equivalent.

Reply:
The `command-line-api-3` example I've added takes over object initialization from the model binder, which shaves about 10ms off of the previous project (still included for comparison). This API could also use a small improvement so that the lookup isn't name based, which is already supported in other APIs.